### PR TITLE
ci(test): use matrix strategy to test against all maintained node versions

### PR DESCRIPTION
**Pull request by shaftoe**
Ref: https://github.com/probot/create-probot-app/issues/309#issuecomment-754871569

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -12,7 +12,10 @@ jobs:
...
       - uses: actions/checkout@v2
-      - uses: actions/setup-node@v1
+      - uses: actions/setup-node@v2
         with:
-          node-version: 12
+          node-version: ${{ matrix.node }}
       - run: npm test
+    strategy:
+      matrix:
+        node: [10, 12, 14, 15]
```
Comment (gr2m) [inline .github/workflows/test.yml:21]: could you move `strategy` above `steps`? I think that order makes more sense. Other than that this looks good 👍🏼

**Review by gr2m — APPROVED** [review]
Thanks!

**Comment by gr2m** [comment]
Ah, I forgot. Once we test against a test matrix, it's somewhat hard to keep the branch protection requirements up-to-date:

image

What I usually do is add a 2nd job, which runs only if the test matrix succeeds. To make tests run a bit faster, I only run the tests, without the linting, as part of the matrix. Then in the 2nd job I run linting

You can see it in action here:
https://github.com/probot/probot/blob/59f07c9bfcb0422e452eab267d4a4170b1f54d35/.github/workflows/test.yml#L13-L52

Could you add that please?

**Comment by shaftoe** [comment]
What is 
Screenshot 2021-01-06 at 10 07 39
referring to?

**Comment by gr2m** [comment]
> What is
> Screenshot 2021-01-06 at 10 07 39
> referring to?

It's still configured in the branch protection configuration as a required status

image

**Comment by shaftoe** [comment]
> It's still configured in the branch protection configuration as a required status

I'm still confused, is it something still missing in the patch or is it doing what you asked for?

**Comment by gr2m** [comment]
I just renamed the job name from `lint` to `test` so we can keep the `test` name in the branch protection settings: 7fa654e

We only do lint in it now, but we could do other tasks that are independent from the test matrix above in future. 

Sorry for the confusion. I hope it makes sense now? If not let me know, I'm happy to explain it more, I think it's a very useful hack to lower maintenance overhead when working with nodejs version matrix & branch protection

**Comment by shaftoe** [comment]
Yeah now makes total sense, I guess my confusion was coming from the overloaded "test" word, somehow I didn't link it to the actual test title in the yaml 😅

**Pull request merged**